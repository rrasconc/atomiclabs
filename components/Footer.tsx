import { View, StyleSheet } from 'react-native';

import { SocialMediaButton } from './Social.Media.Button';
import { Text } from './Text';
import { COLORS, FONT, FOOTER_HEIGHT, SPACING } from '../constants/theme';

export function Footer() {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Â© 2020 AtomicLabs. Todos los derechos reservados.</Text>
      <Text style={[styles.text, styles.anchor]}>Aviso de privacidad</Text>

      <View style={styles.socialMediaContainer}>
        <SocialMediaButton icon="linkedin" url="https://www.linkedin.com/company/atomic-32/" />
        <SocialMediaButton icon="twitter" url="https://twitter.com/Atomic_32" />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: SPACING.medium,
    backgroundColor: COLORS.black,
    justifyContent: 'center',
    alignItems: 'center',
    gap: SPACING.medium,
    height: FOOTER_HEIGHT,
    position: 'absolute',
    bottom: 0,
    right: 0,
    left: 0,
  },
  text: {
    fontSize: FONT.extraSmall,
  },
  anchor: {
    textDecorationLine: 'underline',
  },
  socialMediaContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: SPACING.medium,
  },
});
