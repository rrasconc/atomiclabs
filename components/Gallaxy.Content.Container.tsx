import { useHeaderHeight } from '@react-navigation/elements';
import { View, StyleSheet, ViewProps } from 'react-native';

import { FOOTER_HEIGHT, SPACING } from '../constants/theme';
export function GallaxyContentContainer({ ...viewProps }: ViewProps) {
  const headerHeight = useHeaderHeight();

  return (
    <View
      {...viewProps}
      style={[styles.container, { paddingTop: headerHeight + SPACING.medium }]}
    />
  );
}

const styles = StyleSheet.create({
  container: {
    zIndex: 2,
    flex: 1,
    padding: SPACING.medium,
    gap: SPACING.medium,
    paddingBottom: FOOTER_HEIGHT,
  },
});
