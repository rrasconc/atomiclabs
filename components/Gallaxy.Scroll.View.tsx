import { useNavigation } from '@react-navigation/native';
import { LegacyRef, forwardRef, useState } from 'react';
import {
  type NativeScrollEvent,
  type NativeSyntheticEvent,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';

import { COLORS } from '../constants/theme';
import { GallaxyScrollViewProps } from '../constants/types';

//Scrollview with a background image of a gallaxy
export const GallaxyScrollView = forwardRef(function GallaxyScrollView(
  { children, renderBgImages, ...otherProps }: GallaxyScrollViewProps,
  ref: LegacyRef<ScrollView>
) {
  const navigation = useNavigation();
  const [prevScrollY, setPrevScrollY] = useState(0);

  const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
    let backgroundColor = 'transparent';
    const currentScrollY = event.nativeEvent.contentOffset.y;

    if (currentScrollY > 0 && currentScrollY > prevScrollY) {
      backgroundColor = COLORS.background;
    }

    navigation.setOptions({
      headerStyle: {
        backgroundColor,
      },
    });
    setPrevScrollY(currentScrollY);
  };

  return (
    <ScrollView onScroll={handleScroll} {...otherProps} ref={ref} style={styles.container}>
      <View style={styles.bgContainer}>{renderBgImages()}</View>
      {children}
    </ScrollView>
  );
});

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: COLORS.background,
  },
  bgImage: {
    zIndex: 1,
    position: 'absolute',
  },
  bgContainer: {
    zIndex: 1,
    ...StyleSheet.absoluteFillObject,
    alignItems: 'center',
  },
});
