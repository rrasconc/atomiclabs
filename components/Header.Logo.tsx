import { StyleSheet, Image } from 'react-native';
import { scale, verticalScale } from 'react-native-size-matters';

import { IMAGES } from '../constants/files';

export function HeaderLogo() {
  return <Image style={styles.logo} source={IMAGES.logo} />;
}

const styles = StyleSheet.create({
  logo: {
    height: verticalScale(28),
    resizeMode: 'contain',
    width: scale(200),
  },
});
