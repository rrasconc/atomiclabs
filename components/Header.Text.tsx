import { Text, StyleSheet, TextProps } from 'react-native';
import { COLORS, FONT } from '../constants/theme';

export function HeaderText({ style: customStyle, ...otherProps }: TextProps) {
  const style = [styles.text, customStyle];

  return <Text adjustsFontSizeToFit={true} numberOfLines={1} style={style} {...otherProps} />;
}

const styles = StyleSheet.create({
  text: {
    color: COLORS.white,
    fontSize: FONT.large,
    fontWeight: '700',
    textAlign: 'center',
  },
});
