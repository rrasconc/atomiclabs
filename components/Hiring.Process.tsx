import { View, StyleSheet, Dimensions, Image } from 'react-native';

import { HeaderText } from './Header.Text';
import { COLORS, SPACING } from '../constants/theme';

const peopleSrc = require('../assets/people.png');
const peopleRatio = 416 / 1209;

const { width } = Dimensions.get('window');

export function HiringProcess() {
  return (
    <View style={styles.container}>
      <View>
        <HeaderText>¡TE ENCANTARÁ</HeaderText>
        <HeaderText style={styles.highlightText}> TRABAJAR CON</HeaderText>
        <HeaderText style={styles.highlightText}>NOSOTROS!</HeaderText>
      </View>

      <Image resizeMode="contain" style={styles.peopleImg} source={peopleSrc} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    gap: SPACING.large * 2,
    marginVertical: SPACING.large,
  },
  peopleImg: {
    alignSelf: 'center',
    width: '100%',
    height: width * peopleRatio,
  },
  highlightText: {
    color: COLORS.orange,
  },
});
