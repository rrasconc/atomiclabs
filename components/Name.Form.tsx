import { View, StyleSheet } from 'react-native';

import { PrimaryButton } from './Primary.Button';
import { Text } from './Text';
import { TextInput } from './Text.Input';
import { globalStyles } from '../constants/styles';
import { SPACING } from '../constants/theme';
import { NameFormProps } from '../constants/types';

export function NameForm({
  errors,
  handleChange,
  values: { name, lastname },
  handleSubmit,
  loading,
}: NameFormProps) {
  const isFormPending = !name || !lastname;
  const hasErrors = errors.name !== undefined || errors.lastname !== undefined;

  return (
    <View style={styles.container}>
      <Text style={globalStyles.label}>
        Queremos saber que eres t√∫, por favor ingresa los siguientes datos:
      </Text>
      <TextInput
        error={errors.name ?? ''}
        onChangeText={handleChange('name')}
        value={name}
        label="Nombre (s)"
      />
      <TextInput
        error={errors.lastname ?? ''}
        onChangeText={handleChange('lastname')}
        value={lastname}
        label="Apellidos (s)"
      />
      <PrimaryButton
        disabled={isFormPending || hasErrors || loading}
        onPress={() => handleSubmit()}
        text="Enviar"
        loading={loading}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    gap: SPACING.large,
  },
});
