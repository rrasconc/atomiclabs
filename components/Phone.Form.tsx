import { View, StyleSheet } from 'react-native';

import { PrimaryButton } from './Primary.Button';
import { Text } from './Text';
import { TextInput } from './Text.Input';
import { globalStyles } from '../constants/styles';
import { SPACING } from '../constants/theme';
import { PhoneFormProps } from '../constants/types';

export function PhoneForm({
  errors,
  handleChange,
  values: { phoneNumber },
  handleSubmit,
  loading,
}: PhoneFormProps) {
  return (
    <View style={styles.container}>
      <Text style={globalStyles.label}>Necesitamos validar tu número para continuar</Text>
      <Text style={globalStyles.label}>
        Ingresa tu número a 10 dígitos y te enviaremos un código SMS
      </Text>

      <TextInput
        error={errors.phoneNumber ?? ''}
        onChangeText={handleChange('phoneNumber')}
        value={phoneNumber}
        label="Número de celular"
      />
      <PrimaryButton
        disabled={!phoneNumber || errors.phoneNumber !== undefined || loading}
        onPress={() => handleSubmit()}
        text="Continuar"
        loading={loading}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    gap: SPACING.large,
  },
});
