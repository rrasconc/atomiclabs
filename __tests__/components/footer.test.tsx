import { render, screen } from '@testing-library/react-native';

import { Footer } from '../../components/Footer';

describe('<Footer />', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<Footer />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows copyright label', () => {
    expect(screen.getByText(/atomicLabs. todos los derechos reservados/i)).toBeDefined();
  });

  it('shows privacy policy', () => {
    expect(screen.getByText(/aviso de privacidad/i)).toBeDefined();
  });
});
