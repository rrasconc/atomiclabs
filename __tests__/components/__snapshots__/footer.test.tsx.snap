// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "bottom": 0,
      "gap": 32.14285714285714,
      "height": 274.6470588235294,
      "justifyContent": "center",
      "left": 0,
      "paddingHorizontal": 32.14285714285714,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 38.57142857142857,
        },
        {
          "fontSize": 27.857142857142858,
        },
      ]
    }
  >
    © 2020 AtomicLabs. Todos los derechos reservados.
  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 38.57142857142857,
        },
        [
          {
            "fontSize": 27.857142857142858,
          },
          {
            "textDecorationLine": "underline",
          },
        ],
      ]
    }
  >
    Aviso de privacidad
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 32.14285714285714,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={21.428571428571427}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 47.08235294117647,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={21.428571428571427}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 47.08235294117647,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
